syncd/PortStateChangeHandler.cpp

```diff
@@ -0,0 +1,46 @@
+#include "PortStateChangeHandler.h"
+
+#include <string>
+
+using namespace syncd;
+
+PortStateChangeHandler::PortStateChangeHandler(
+        _In_ std::shared_ptr<swss::SelectableEvent> portStateChangeEvent)
+    : m_portStateChangeEvent(portStateChangeEvent)
+{
+    SWSS_LOG_ENTER();
+
+    m_portStateChangeQueue = std::make_shared<portOperStatusNotificationQueue>(
+            m_portStateChangeQueueSize);
+}
+
+PortStateChangeHandler::~PortStateChangeHandler()
+{
+    SWSS_LOG_ENTER();
+}
+
+void PortStateChangeHandler::handlePortStateChangeNotification(
+        _In_ uint32_t count,
+        _In_ const sai_port_oper_status_notification_t *data)
+{
+    SWSS_LOG_ENTER();
+
+    if (m_portStateChangeEvent == nullptr)
+    {
+        SWSS_LOG_THROW("Unexpected error: port state change event is null.");
+    }
+
+    for (uint32_t idx = 0; idx < count; ++idx)
+    {
+        if (m_portStateChangeQueue->enqueue(data[idx]) == false)
```

Comment:
why this can fail? im suspecting that you added limit to queue size, are you expecting like notification storm or something like that the will exceed notification queue ?

Reply:
The queue size should be enough but check is for just in case queue is filled up due to some issue( for example: link event damper is not able to clean up the port events from queue due to some software issue).